src/osx/carbon/statbrma.cpp

```diff
@@ -147,8 +148,24 @@ void wxStatusBarMac::OnPaint(wxPaintEvent& WXUNUSED(event))
     int w, h;
     GetSize( &w, &h );
 
+    // Notice that wxOSXGetKeyWindow (aka [NSApp keyWindow] used below is
+    // subtly different from IsActive() (aka [NSWindow iskeyWindow]): the
+    // former remains non-NULL if another application shows a temporary
+    // floating window or a status icon's menu is used. That's what we want: in
+    // that case, statusbar appearance shouldn't change. It also shouldn't
+    // change if a window-modal sheet attached to this window is key.
     wxTopLevelWindow *tlw = wxDynamicCast(MacGetTopLevelWindow(), wxTopLevelWindow);
-    if ( tlw && tlw->IsActive() )
+    wxWindow *keyWindow = wxNonOwnedWindow::GetFromWXWindow(wxOSXGetKeyWindow())->MacGetTopLevelWindow();
+    while ( keyWindow && keyWindow != tlw )
+    {
+        wxDialog *dlg = wxDynamicCast(keyWindow, wxDialog);
```

Comment:
I thought it better not to add yet another to the many helpers already in there, because I don’t think it would be useful for anything else.